lib/dom.js

```diff
@@ -1188,6 +1191,23 @@ function cloneNode(doc,node,deep){
 	return node2;
 }
 
+function getTextContent(node){
+	switch(node.nodeType){
+	case ELEMENT_NODE:
+	case DOCUMENT_FRAGMENT_NODE:
+		var buf = [];
+		node = node.firstChild;
+		while(node){
+			if(node.nodeType!==7 && node.nodeType !==8){
+				buf.push(getTextContent(node));
+			}
+			node = node.nextSibling;
+		}
+		return buf.join('');
+	default:
+		return node.nodeValue;
+	}
+}
```

Comment:
What is the reason for moving this function here? At first glance is seems to be unrelated to the linting changes. If so you could extract it into a separate PR.

Reply:
This function was previously inside an "if" statement, which is ambiguous behavior because of function hoisting; this is prohibited in strict mode.